src/components/atoms/SupplierCoupons/index.tsx

```diff
@@ -1,61 +1,122 @@
 import ShoppingCartGreen from "@/images/easytolive/icons/shopping_cart_green.svg";
 import ClockCircleRed from "@/images/easytolive/icons/clock_circleRed.svg";
 import classNames from "@/utils/classNames";
 
 interface SupplierCouponsProps {
+  title: string;
   discount: string;
   unintsAmount: number;
-  expirateTime: number;
+  expirateTime: string;
   icon: string | StaticImageData;
+  status?: "active" | "used" | "expired";
+  usageDate?: string;
 }
 
 const SupplierCoupons: React.FC<SupplierCouponsProps> = ({
+  title,
   discount,
   unintsAmount,
   expirateTime,
   icon,
+  usageDate,
+  status = "active",
 }) => {
-  return (
-    <div className="bg-primary-main h-auto pl-4 rounded-full flex items-center gap-3 cursor-pointer">
-      <h2 className=" font-bold text-white text-xl">{discount}%</h2>
+  const STATUS_COMPONENTS = {
+    used: (
+      <div>
+        <p className="text-xs pb-2 font-semibold text-black">{title}</p>
+        <div className="flex flex-col gap-0.5 text-xs">
+          <p className="flex font-semibold items-center text-generic-alertGreen">
+            <Image
+              className="h-auto pr-2 w-7"
+              alt="coupon-black"
+              src={CouponGreen}
+            />
+            utilizado em {usageDate}
+          </p>
+        </div>
+        <span></span>
+      </div>
+    ),
+    expired: (
+      <div>
+        <p className="text-xs pb-2 font-semibold text-black">{title}</p>
+        <div className="flex flex-col gap-1 text-xs">
+          <p className="flex font-semibold items-center text-generic-alertRed">
+            <Image
+              className="h-auto pr-2 w-7"
+              alt="coupon-black"
+              src={CouponRed}
+            />
+            expirou em {expirateTime}
+          </p>
+        </div>
+        <span></span>
+      </div>
+    ),
+    active: (
+      <div className="flex items-center justify-between gap-4">
+        <div className="flex flex-col gap-0.5 text-xs">
+          <p className="text-xs font-semibold text-black">{title}</p>
+          <p className="flex font-semibold items-center text-generic-alertGreen">
+            <Image
+              className="h-3.5 pr-2 w-auto"
+              alt="coupon-black"
+              src={ShoppingCartGreen}
+              color="white"
+            />
+            faltam {unintsAmount} unidades
+          </p>
+          <p className="flex font-semibold items-center text-generic-alertRed">
+            <Image
+              className="h-3.5 pr-2 w-auto"
+              alt="coupon-black"
+              src={ClockCircleRed}
+            />
+            termina em {expirateTime}
+          </p>
+        </div>
+        <Image className="h-6 w-auto" alt="arrow right" src={icon} />
+      </div>
+    ),
+  };
 
+  return (
+    <div
+      className={`${
+        status === "used"
+          ? "bg-secondary-main"
+          : status === "expired"
+          ? "bg-generic-alertRed"
+          : "bg-primary-main"
```

Comment:
This is wrong. We should NOT use multiples ternaries

Reply:
Try using a different approach using an object or a function to find the background-color according to the status.

Option 1:

const BG_COLORS_BY_STATUS = {
  used: 'bg-secondary-main',
  expired 'bg-generic-alertRed',
  ....
}



Option 2: 

const getBgColorByStatus = (status) => { ... }

Reply:
Remember: if you need to do `if, else if, else if, else if....`, the ternary is not the best option. It will if it's only: `if, else`